client-side/src/components/profileComponents/timerActivationButton.jsx

```diff
@@ -0,0 +1,113 @@
+    const [start, setStart] = useState('00:00');
+    const [end, setEnd] = useState('00:00');
+    const [timers, setTimers] = useState([]);
+
+    const handleClickOpen = useCallback(() => {
+        setOpen(true);
+    }, []);
+
+    const handleClose = useCallback(() => {
+        setOpen(false);
+    }, []);
+
+    function parseTimeStringToDate(timeString) {
+        const [hours, minutes] = timeString.split(':').map(Number);
+        const date = new Date();
+        date.setHours(hours, minutes, 0, 0);
+        return date;
+    }
+
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        if (name === 'timeStart') {
+            setStart(value);
+        } else if (name === 'timeEnd') {
+            setEnd(value);
+        }
+    }
+
+    const handleStart = () => {
+        handleClose();
+
+        const s = parseTimeStringToDate(start);
+        const e = parseTimeStringToDate(end);
+        const durationMinutes = (e - s) / 1000 / 60;
+        const duration = Math.abs(durationMinutes);
+
+        const existingTimerIndex = timers.findIndex(timer => timer.profileName === profileName);
+        
+        if (existingTimerIndex >= 0) {
+            const updatedTimers = [...timers];
+            updatedTimers[existingTimerIndex] = { profileName, duration };
+            setTimers(updatedTimers);
+        } else {
+            setTimers([...timers, { profileName, duration }]);
+        }
+    }
+
+    return (
+        <div>
+            <GenericButton label={BUTTON_LABELS.TIMER_ACTIVATION} onClick={handleClickOpen} size="medium" className="profile-timer-button" />
+            <Dialog open={open} onClose={handleClose} fullWidth={true}>
+                <DialogTitle>{DIALOG_TITLES.TIMER_ACTIVATION}</DialogTitle>
+                <DialogContent>
+                    <Grid container spacing={3} mt={2}>
+                        <Grid item xs={12} sm={3}>
+                            <GenericInput
+                                label={INPUT_LABELS.TIME_START}
+                                name="timeStart"
+                                value={start}
+                                onChange={handleChange}
+                                type="time"
+                                width='100%'
+                            />
+                        </Grid>
+                        <Grid item xs={12} sm={3}>
+                            <GenericInput
+                                label={INPUT_LABELS.TIME_END}
+                                name="timeEnd"
+                                value={end}
+                                onChange={handleChange}
+                                type="time"
+                                width='100%'
+                            />
+                        </Grid>
+                    </Grid>
+                </DialogContent>
+                <DialogActions>
+                    <Button sx={{ color: 'rgb(103, 252, 210)' }} onClick={handleClose}>
```

Comment:
colors shouldn't be defined in jsx files, they should be defined in variables file and used in scss file